src/main/resources/META-INF/resources/primefaces/datatable/datatable.js

```diff
@@ -2396,8 +2402,9 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
      * @param {JQuery.TriggeredEvent} event Event that occurred.
      * @param {JQuery} rowElement Row that was clicked.
      * @param {PrimeFaces.widget.DataTable.CmSelectionMode} cmSelMode The current selection mode.
+     * @param {function} fnShowMenu callback to open menu.
```

Comment:
@naofireblade You need to specify the signature of the callback function (as a TypeScript type). In this case, from what I can see, I would guess it should be

```
     * @param {() => void} [fnShowMenu] Optional callback function invoked when the menu was opened.
``` 

The `[...]` indicate that the parameter is optional.

Same goes for the `fireRowSelectEvent` method

Reply:
I'm new to ts and had no idea how to classify a callback function in the parameter documentation.

Reply:
Yeah no problem we're not blaming you : )

Reply:
Thanks for your help! Will fix it later today.